src/NServiceBus.Testing.Fakes/TestableRecoverabilityContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace NServiceBus.Testing
+{
+    using System;
+    using System.Collections.Generic;
+    using NServiceBus.Extensibility;
+    using Pipeline;
+    using Transport;
+
+    /// <summary>
+    /// A testable implementation of <see cref="IRecoverabilityContext" />.
+    /// </summary>
+    public partial class TestableRecoverabilityContext : TestableBehaviorContext, IRecoverabilityContext
+    {
+        /// <summary>
+        /// The message that failed processing.
+        /// </summary>
+        public ErrorContext ErrorContext { get; set; } = new ErrorContext(
+            new Exception(),
+            new Dictionary<string, string>(),
+            Guid.NewGuid().ToString(),
+            ReadOnlyMemory<byte>.Empty,
+            new TransportTransaction(),
+            0,
+            "",
+            new ContextBag());
+
+        /// <summary>
+        /// The recoverability action to take for the failed message.
+        /// </summary>
+        public ErrorHandleResult ActionToTake { get; set; }
```

Comment:
If we would also expose the recoverability settings on the context could we move the recoverability strategy etc into a terminator or even replaceable behaviour if we want to fancy?